synnax/pkg/distribution/channel/retrieve.go

```diff
@@ -1,137 +1,149 @@
+
+	"github.com/cockroachdb/errors"
+	"github.com/samber/lo"
+	"github.com/synnaxlabs/synnax/pkg/distribution/core"
+	"github.com/synnaxlabs/synnax/pkg/distribution/ontology"
+	"github.com/synnaxlabs/synnax/pkg/distribution/ontology/search"
+	"github.com/synnaxlabs/x/gorp"
+	"github.com/synnaxlabs/x/query"
+)
+
+// Retrieve is used to retrieve information about Channel(s) in delta's distribution
+// layer.
+type Retrieve struct {
+	tx                        gorp.Tx
+	gorp                      gorp.Retrieve[Key, Channel]
+	otg                       *ontology.Ontology
+	keys                      Keys
+	searchTerm                string
+	validateRetrievedChannels func(channels []Channel) ([]Channel, error)
+}
+
+func newRetrieve(tx gorp.Tx, otg *ontology.Ontology, validateRetrievedChannels func([]Channel) ([]Channel, error)) Retrieve {
+	return Retrieve{
+		gorp:                      gorp.NewRetrieve[Key, Channel](),
+		tx:                        tx,
+		otg:                       otg,
+		validateRetrievedChannels: validateRetrievedChannels,
+	}
+}
+
+func (r Retrieve) Search(term string) Retrieve { r.searchTerm = term; return r }
+
+// Entry binds the Channel that Retrieve will fill results into. This is an identical
+// interface to gorp.Retrieve.
+func (r Retrieve) Entry(ch *Channel) Retrieve { r.gorp.Entry(ch); return r }
+
+// Entries binds a slice that Retrieve will fill results into.  This is an identical
+// interface to gorp.Retrieve.
+func (r Retrieve) Entries(ch *[]Channel) Retrieve { r.gorp.Entries(ch); return r }
+
+// WhereNodeKey filters for channels whose Leaseholder attribute matches the provided
+// leaseholder node Key.
+func (r Retrieve) WhereNodeKey(nodeKey core.NodeKey) Retrieve {
+	r.gorp.Where(func(ch *Channel) bool { return ch.Leaseholder == nodeKey })
+	return r
+}
+
+// WhereNames filters for channels whose Key attribute matches the provided name.
+func (r Retrieve) WhereNames(names ...string) Retrieve {
+	matchers := make([]func(string) bool, len(names))
+	for i, name := range names {
+		matchers[i] = formatNameMatcher(name)
+	}
+	r.gorp.Where(func(ch *Channel) bool {
+		return lo.SomeBy(matchers, func(matcher func(string) bool) bool {
+			return matcher(ch.Name)
+		})
+	})
+	return r
+}
+
+// WhereKeys filters for channels with the provided Key. This is an identical interface
+// to gorp.Retrieve.
+func (r Retrieve) WhereKeys(keys ...Key) Retrieve {
+	r.keys = append(r.keys, keys...)
+	r.gorp.WhereKeys(keys...)
+	return r
+}
+
+// Limit limits the number of results returned by the query. This is an identical
+// interface to gorp.Retrieve.
+func (r Retrieve) Limit(limit int) Retrieve { r.gorp.Limit(limit); return r }
+
+// Offset offsets the results returned by the query. This is an identical interface to
+// gorp.Retrieve.
+func (r Retrieve) Offset(offset int) Retrieve {
+	r.gorp.Offset(offset)
+	return r
+}
+
+// Exec executes the query, binding
+func (r Retrieve) Exec(ctx context.Context, tx gorp.Tx) (err error) {

```

Comment:
Why changed to named return? IMO if a function is named return then it's return should not have any arguments, where possible